pkg/controllers/osc/osc_reconciler_test.go

```diff
@@ -20,75 +20,91 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"os"
 	"reflect"
 	"testing"
+	"time"
 
 	"github.com/kubermatic/machine-controller/pkg/apis/cluster/v1alpha1"
 	providerconfigtypes "github.com/kubermatic/machine-controller/pkg/providerconfig/types"
 
-	"k8c.io/operating-system-manager/pkg/controllers/osc/resrources"
+	"k8c.io/operating-system-manager/pkg/controllers/osc/resources"
 	osmv1alpha1 "k8c.io/operating-system-manager/pkg/crd/osm/v1alpha1"
 	"k8c.io/operating-system-manager/pkg/generator"
+	testUtil "k8c.io/operating-system-manager/pkg/test/util"
 
 	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/utils/pointer"
 	fakectrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var (
+	osmProfile = &osmv1alpha1.OperatingSystemProfile{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "ubuntu-20.04-profile",
+			Namespace: "kube-system",
+		},
+		Spec: osmv1alpha1.OperatingSystemProfileSpec{
+			OSName:    "Ubuntu",
+			OSVersion: "20.04",
+			Files: []osmv1alpha1.File{
+				{
+					Path:        "/opt/bin/setup",
+					Permissions: pointer.Int32Ptr(0755),
+					Content: osmv1alpha1.FileContent{
+						Inline: &osmv1alpha1.FileContentInline{
+							Encoding: "b64",
+							Data:     "#!/bin/bash\nset -xeuo pipefail\ncloud-init clean\nsystemctl start provision.service",
+						},
+					},
+				},
+				{
+					Path:        "/etc/systemd/system/setup.service",
+					Permissions: pointer.Int32Ptr(0644),
+					Content: osmv1alpha1.FileContent{
+						Inline: &osmv1alpha1.FileContentInline{
+							Encoding: "b64",
+							Data:     "[Install]\nWantedBy=multi-user.target\n\n[Unit]\nRequires=network-online.target\nAfter=network-online.target\n[Service]\nType=oneshot\nRemainAfterExit=true\nExecStart=/opt/bin/setup",
+						},
+					},
+				},
+			},
+		},
+	}
+	// Path for a dummy kubeconfig; not using a real kubeconfig for this use case
+	kubeconfigPath    = os.Getenv("PWD") + "/../../../testdata/kube-config.yaml"
```

Comment:
please create a dir in this package called tesdata and in here add the file. Your mock should be in the same package that you are testing. Easier to maintain.